src/util/math.coffee

```diff
@@ -12,6 +12,28 @@ module.exports.MAX_DATE_VALUE = MAX_DATE_VALUE = DateTime.parse("9999-12-31T23:5
 module.exports.MIN_TIME_VALUE = MAX_TIME_VALUE = DateTime.parse("0000-01-01T00:00:00.000")
 module.exports.MAX_TIME_VALUE = MAX_TIME_VALUE = DateTime.parse("0000-01-01T23:59:59.999")
 
+
+module.exports.isValidInteger = isValidInteger = (integer) ->
+  throw new Error("Unable to parse Integer") if isNaN(integer)
+  throw new Error("Maximum Integer value exceeded") if integer > MAX_INT_VALUE
+  throw new Error("Minimum Integer value exceeded") if integer < MIN_INT_VALUE
+  return true
+
+module.exports.isValidDecimal = isValidDecimal = (decimal) ->
+  throw new Error("Unable to parse Decimal") if isNaN(decimal)
+  throw new Error("Maximum Decimal value exceeded") if decimal > MAX_FLOAT_VALUE
+  throw new Error("Minimum Decimal value exceeded") if decimal < MIN_FLOAT_VALUE
+  throw new Error("Maximum Decimal precision") if getDecimalPrecision(decimal) > 8
```

Comment:
This one kind of concerns me.  If we do math, we may get a decimal with greater precision than 8 digits.  I don't think that's an error condition / invalid decimal.  Perhaps we could truncate at 8 digits of precision, but I wouldn't want to throw an error.

I actually was kind of choosing to interpret the 8 digits of precision as a requirement to support at least 8 digits of precision, and that when `successor` / `predecessor` is used, we should move up or down to the next 8th digit of precision.

Reply:
Good point.  I think it would be good to truncate in order to align to this clause in 9.3.9:

> Note that the decimal value returned by this operator must be limited in precision and scale to the maximum precision and scale representable for Decimal values within CQL.

Reply:
Ha.  I guess I should have checked the spec.  Agreed.